torch/_dynamo/eval_frame.py

```diff
@@ -884,6 +887,10 @@ def graph_with_interpreter(*args):
         graph,
     ).transform()
 
+    new_graph.meta["example_inputs"] = example_inputs
+    new_graph.meta["input_shape_constraints"] = constraints
+    new_graph.meta["node_range_constraints"] = node_range_constraints
```

Comment:
It looks like what gets stored here is a dup of input constraints
```
input_shape_constraints:  [
    Constraint(w_tensor=<weakref at 0x7f2ac6eef3d0; to 'Tensor' at 0x7f2ad64c7f60>, t_id=139821960691552, dim=0, constraint_range=StrictMinMaxConstraint(warn_only=False, vr=ValueRanges(lower=2, upper=oo))), 
    Constraint(w_tensor=<weakref at 0x7f2ac6dae890; to 'Tensor' at 0x7f2ac7fb5f30>, t_id=139821720493872, dim=0, constraint_range=StrictMinMaxConstraint(warn_only=False, vr=ValueRanges(lower=2, upper=oo))), 
    Constraint(w_tensor=<weakref at 0x7f2ac6daea70; to 'Tensor' at 0x7f2ac7fb5030>, t_id=139821720490032, dim=0, constraint_range=StrictMinMaxConstraint(warn_only=False, vr=ValueRanges(lower=2, upper=oo)))
]
node_range_constraints:  {
    s0: ValueRanges(lower=2, upper=9223372036854775806), 
    s1: ValueRanges(lower=2, upper=9223372036854775806), 
    s2: ValueRanges(lower=2, upper=9223372036854775806)
}
```

When I check the id of underlying `ValueRanges` object gets stored, they are different. I assume they should be identical

Reply:
Why store `var_to_range`, then? I'm not surprised that the ids are different, because we do `&` on value ranges everywhere. Also, in general input constraints will subsume this information.

Reply:
@guangy10 could you try saving/loading the graph module produced from dynamo after your change? I remember there are some complications when we use weakref inside a graph module for other users who call save/load on torchdynamo.export graph. If it doesnt work, you can just delete the weakref. I will use tensor_id and example inputs to do the matching.

Reply:
And you can delete the constraints on placeholder variables as they will be subsumed by the input constraints

Reply:
> could you try saving/loading the graph module produced from dynamo after your change? I remember there are some complications when we use weakref inside a graph module for other users who call save/load on torchdynamo.export graph. If it doesnt work, you can just delete the weakref. I will use tensor_id and example inputs to do the matching.

@tugsbayasgalan As you said, save/load doesn't work with weakref inside graph module. 
```
TypeError: cannot pickle 'weakref.ReferenceType' object
```
So I will delete each weakref prior to saving the constaints as metadata